src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -874,6 +875,19 @@ await CallWithRequestTracing(async () =>
 
                             foreach (ConfigurationSetting setting in page.Values)
                             {
+                                if (setting.ContentType == SnapshotReferenceConstants.ContentType)
+                                {
+                                    SnapshotReference snapshotReference = new SnapshotReference(SnapshotReferenceParser.ParseSnapshotName(setting));
+                                    Dictionary<string, ConfigurationSetting> resolvedSettings = await Resolve(snapshotReference, client, cancellationToken).ConfigureAwait(false);
```

Comment:
If you pass `data` dictionary to the helper function, you can update it in the helper method directly instead of doing it here.
The logic here can be simplified to the following:
```
if (setting.ContentType == SnapshotReferenceConstants.ContentType && setting.Value != null && setting.Value.TryParseSnapshotName(out string snapshotName))
{
    await LoadSnapshotData(snapshotName, data, client, cancellationToken).ConfigureAwait(false);

    continue;
}
```

Reply:
Since we're not adding `TryParseSnapshotName`, this gets updated to:

```
if (setting.ContentType == SnapshotReferenceConstants.ContentType)
{
    await LoadSnapshotData(SnapshotReferenceParser.Parse(setting), data, client, cancellationToken).ConfigureAwait(false);

    continue;
}
```